query/src/main/scala/filodb/query/exec/MetadataExecPlan.scala

```diff
@@ -78,13 +82,17 @@ final case class LabelValuesDistConcatExec(id: String,
       case QueryResult(_, _, result) => result
       case QueryError(_, ex)         => throw ex
     }.toListL.map { resp =>
-      var metadataResult = Seq.empty[ZeroCopyUTF8String]
+      var metadataResult = scala.collection.mutable.Set.empty[Map[ZeroCopyUTF8String, ZeroCopyUTF8String]]
       resp.foreach(rv => {
-        metadataResult ++= rv(0).rows.toSeq.map(rowReader => rowReader.filoUTF8String(0))
+        metadataResult ++= rv(0).rows.toSeq.map(rowReader => {
+          val binaryRowReader = rowReader.asInstanceOf[BinaryRecordRowReader]
+          rv(0).schema.toStringPairs(binaryRowReader.recordBase, binaryRowReader.recordOffset)
+            .map(pair => pair._1.utf8 -> pair._2.utf8).toMap
+        })
       })
       //distinct -> result may have duplicates in case of labelValues
       IteratorBackedRangeVector(new CustomRangeVectorKey(Map.empty),
-        new ZCUTF8IteratorRowReader(metadataResult.distinct.toIterator))
+        new UTF8MapIteratorRowReader(metadataResult.toIterator))
```

Comment:
Today we have the same limit for samples and metadata results. I am thinking we send in different limits for metadata queries.